File: Server/Server/Server/Model/DTO/UpdateUserDTO.cs
namespace Server.Model.DTO
{
    /// <summary>
    /// Représente les données de modification d'un utilisateur.
    /// </summary>
    public class UpdateUserDTO
    {
        private string tokenuser;
        private string? username;
        private string? email;
        private string oldpassword;
        private string? password;
        private IFormFile? profilePic;

        /// <summary>
        /// Le token associé à l'utilisateur connecté souhaitant modifier ses informations.
        /// </summary>
        public string Tokenuser { get => tokenuser; set => tokenuser = value; }

        /// <summary>
        /// Le nouveau nom d'utilisateur de l'utilisateur connecté.
        /// </summary>
        public string? Username { get => username; set => username = value; }

        /// <summary>
        /// Le nouvel email de l'utilisateur connecté.
        /// </summary>
        public string? Email { get => email; set => email = value; }

        /// <summary>
        /// Le nouveau mot de passe de l'utilisateur connecté.
        /// </summary>
        public string? Password { get => password; set => password = value; }

        /// <summary>
        /// La nouvelle photo de profil de l'utilisateur connecté.
        /// </summary>
        public IFormFile? ProfilePic { get => profilePic; set => profilePic = value; }

        /// <summary>
        /// L'ancien mot de passe de l'utilisateur 
        /// </summary>
        public string Oldpassword { get => oldpassword; set => oldpassword = value; }
    }
}

File: Client/src/app/Model/DTO/UpdateUserDTO.ts
/**
 * Gère les attributs de l'objet de mise à jour utilisateur (DTO = Data Transfer Object) sert d'objet de transfert de 
 * données entre un client et un serveur
 */
export class UpdateUserDTO {
    private tokenuser: string;
    private username: string;
    private email: string;
    private oldpassword: string
    private password: string;
    private profilePic: File | null;

    /**
     * Renvoi le token de l'utilisateur
     */
    public get TokenUser(): string {
        return this.tokenuser;
    }
    /**
     * Renvoi le nom de l'utilisateur
     */
    public get Username(): string {
        return this.username;
    }
    /**
    * Renvoi l'email de l'utilisateur
    */
    public get Email(): string {
        return this.email;
    }
    
    /**
     * Renvoi l'ancien mot de passe de l'utilisateur
     */
    public get OldPassword(): string {
        return this.oldpassword;
    }

    /**
     * Renvoi le mot de passe de l'utilisateur
     */
    public get Password(): string {
        return this.password;
    }
    /**
     * Renvoi l'image de profil de l'utilisateur
     */
    public get ProfilePic(): File | null {
        return this.profilePic;
    }
  
    constructor(token: string, user: string, mail: string, oldpassword: string, password: string, profile: File) 
    {
        this.tokenuser = token;
        this.username = user;
        this.email = mail;
        this.oldpassword = oldpassword;
        this.password = password;
        this.profilePic = profile;

    }
}
